src/test/java/com/rocketden/main/entity/RoomEntityTests.java

```diff
@@ -62,21 +72,21 @@ public void getEquivalentUserSucceeds() {
         Room room = new Room();
 
         User user = new User();
-        user.setNickname("test");
-        user.setId(1);
+        user.setNickname(NICKNAME);
+        user.setId(ID);
+        user.setUserId(USER_ID);
         room.addUser(user);
 
         User userToGet = new User();
-        userToGet.setNickname("nonexistent");
-        userToGet.setId(2);
+        userToGet.setNickname(NICKNAME_2);
+        userToGet.setId(ID_2);
+        userToGet.setUserId(USER_ID_2);
 
         assertNull(room.getEquivalentUser(userToGet));
 
-        userToGet.setNickname("test");
-
-        User actual = room.getEquivalentUser(userToGet);
+        User actual = room.getEquivalentUser(user);
```

Comment:
nit: Keep the original code where it modifies `userToGet` to have the same `userId`/`nickname` but different id, so that this test can verify that the equivalent user has the same `id` values

So like if `room.users` contains `user{id=1, uid='test'}`, then calling `getEquivalenteUsers` with `otherUser{id=null, uid='test'}` will return the correct object reference of `user{id=1, uid='test'}`

Reply:
Ok, done! I can't see a case where this is actually applicable but it is good to check that our code works as expected, so thanks for this change.